src/main/java/teammates/storage/sqlentity/Notification.java

```diff
@@ -0,0 +1,269 @@
+package teammates.storage.sqlentity;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.util.FieldValidator;
+import teammates.common.util.JsonUtils;
+import teammates.common.util.SanitizationHelper;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+/**
+ * Represents a unique notification in the system.
+ */
+@Entity
+@Table(name = "Notifications")
+public class Notification extends BaseEntity {
+
+    @Id
+    private String notificationId;
```

Comment:
@hhdqirui Should we autogenerate this field instead `@GeneratedValue(strategy = GenerationType.UUID)`. Right now seems like the calling function (that initialises the builder obj) is the one that is responsible for generation of UUID, which may not be what we want.

Reply:
Also we should consider using the UUID type and only typecasting when necessary, might be easier to work with.

Reply:
I agree that using UUID would be better and `@GeneratedValue(strategy = GenerationType.UUID)`. 

> Right now seems like the calling function (that initialises the builder obj) is the one that is responsible for generation of UUID, which may not be what we want.

This is actually what we have right now as defined in `CreateNotificationAction`. But of course, changing it does not really have any impact.